File: 프로그래머스/0/120885. 이진수 더하기/이진수 더하기.py
def solution(bin1, bin2):
    l1 = len(bin1)-1
    a1 =0
    l2 = len(bin2)-1
    a2 = 0
    for i in bin1:
        a1 += int(i)*(2**l1)
        l1 -=1
    for i in bin2:
        a2 += int(i)*(2**l2)
        l2-=1
    b = a1 + a2
    c = format(b, 'b')
    return str(c)

File: 프로그래머스/0/120885. 이진수 더하기/이진수 더하기.java
class Solution {
    public String solution(String bin1, String bin2) {
        String answer = "";
        int len1 = bin1.length() - 1;
        int len2 = bin2.length() - 1;
        int num1 = 0;
        int num2 = 0;
        for (char i : bin1.toCharArray()) {
            num1 += Character.getNumericValue(i) * Math.pow(2, len1);
            len1 -= 1;
        }
        for (char i : bin2.toCharArray()) {
            num2 += Character.getNumericValue(i) * Math.pow(2, len2);
            len2 -= 1;
        }
        int total = num1 + num2;
        answer = Integer.toBinaryString(total);
        return answer;
    }
}